src/esw/imu_driver.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+import rospy
+from sensor_msgs.msg import Imu, MagneticField, Temperature
+from mrover.msg import CalibrationStatus
+
+import serial
+from serial import SerialException
+
+
+def main():
+    # publishers for all types of IMU data, queue size is 1 to make sure we don't publish old data
+    imu_pub = rospy.Publisher("imu", Imu, queue_size=1)
+    mag_pub = rospy.Publisher("magnetometer", MagneticField, queue_size=1)
+    temp_pub = rospy.Publisher("imu_temp", Temperature, queue_size=1)
+    calibration_pub = rospy.Publisher("imu_calibration", CalibrationStatus, queue_size=1)
+
+    # no rate used because rate is effectively set by Arduino serial publisher
+    rospy.init_node("imu_driver")
+
+    # read serial connection info and IMU frame from parameter server
+    port = rospy.get_param("imu_driver/port", "/dev/imu")
```

Comment:
for the yaml file, is there any advantage between explicitly using the ttyACM vs the /dev/imu name?

Reply:
I think using the udev rule for /dev/imu is better since the port never changes, I just don't have that set up on my computer so the ports in the yaml file rn are set to the ACM ones. I'll change that